test/e2e/nodepool_autorepair_test.go

```diff
@@ -0,0 +1,150 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	. "github.com/onsi/gomega"
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/cmd/cluster/core"
+	awsutil "github.com/openshift/hypershift/cmd/infra/aws/util"
+	e2eutil "github.com/openshift/hypershift/test/e2e/util"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func testNodePoolAutoRepair(parentCtx context.Context, mgmtClient crclient.Client, hostedCluster *hyperv1.HostedCluster, hostedClusterClient crclient.Client, clusterOpts core.CreateOptions) func(t *testing.T) {
+	return func(t *testing.T) {
+		g := NewWithT(t)
+
+		ctx, cancel := context.WithCancel(parentCtx)
+		originalNP := hyperv1.NodePool{}
+		defer func() {
+			t.Log("Test: NodePoolAutoRepair finished")
+			cancel()
+		}()
+
+		// List NodePools (should exists only one)
+		nodePools := &hyperv1.NodePoolList{}
+		err := mgmtClient.List(ctx, nodePools, &crclient.ListOptions{
+			Namespace: hostedCluster.Namespace,
+		})
+		g.Expect(err).NotTo(HaveOccurred(), "failed getting existant nodepools")
+		for _, nodePool := range nodePools.Items {
+			if !strings.Contains(nodePool.Name, "-test-") {
+				originalNP = nodePool
+			}
+		}
+		g.Expect(originalNP.Name).NotTo(ContainSubstring("test"))
```

Comment:
You should also have an assertion like:
g.Expect(originalNP.Name).NotTo(BeEmpty())

Otherwise you would not be able to catch the case when a NodePool was not found.